### PR TITLE
pk-main: segfault when config file is not found [v2].

### PR DESCRIPTION
**Pull request by jcfaracco**
When the config file used by PackageKit daemon is not found, the
function `pk_util_get_config_filename()` returns a NULL pointer.

See:
https://github.com/hughsie/PackageKit/blob/master/src/pk-shared.c#L300

So, when `g_key_file_load_from_file()` is called, the variable
`conf_filename` is NULL and inside that function the `assert()` fails:

    $ sudo packagekitd

    (packagekitd:9133): GLib-CRITICAL **: g_key_file_load_from_file:
    assertion 'file != NULL' failed
    Segmentation fault

This commit fixes the problem.

Signed-off-by: Julio Faracco <jfaracco@br.ibm.com>

### CODE REVIEW

File: src/pk-main.c [hunk 1]
```diff
@@ -173,6 +173,10 @@ main (int argc, char *argv[])
 	/* get values from the config file */
 	conf = g_key_file_new ();
 	conf_filename = pk_util_get_config_filename ();
+	if (conf_filename == NULL) {
+		g_print ("Cannot found config file.");
```
Comment (ximion) [inline src/pk-main.c]: Really really minor, but `Cannot found config file.` is weird English - maybe `Could not find config file.` or `Config file not found.` would be better here.
Or, IMHO even better and by using `g_printerr`:
```C
g_printerr ("Config file '%s' not found.", conf_filename);
```

Additionally, but unrelated to this PR: @hughsie, shouldn't we exit with a non-zero exit status if we encounter such initialization issues? At time, PK seems to always terminate with an exit code of zero.

**Comment by ximion** [comment]
Btw, I think you can just force-push to your repo, the bug that Github forgets the comments made on PRs is gone, AFAIK, and the existing PR will just update in that case.

File: src/pk-main.c (hunk 1 above)
Comment (jcfaracco) [inline src/pk-main.c]: Ok, let me change the message. ;-)
I didn't change to `g_printerr()` now to follow the current pattern inside pk-main.c.
I totally agree, but maybe it can be introduced by another patch. There are many occurrences of others `g_print()` for errors in other files either.

**Comment by jcfaracco** [comment]
I create a v3 PR with the recommendations. For reference #190 
Probably, other files will need to be checked if they are using `g_print()` for errors too.

**Comment by Conan-Kudo** [comment]
Can we close this PR, since I think v3 was merged?

**Comment by ximion** [comment]
Jep
